### PR TITLE
Add mask hint for msdeploy when basic auth is disabled

### PR DESCRIPTION
**Pull request by dhashrath91**
Though bearer AuthN is used, lets try to set publish profile password for mask hints to maintain compat with old behavior for MSDEPLOY on best effort basis. 
 This needs to be cleaned up once MSDEPLOY suppport is reomve. Safe handle the exception setting up mask hint as we dont want to fail here.

**Documentation changes required:** (Y/N) <Please mark if documentation changes are required>

**Added unit tests:** (Y/N) <Please mark if unit tests were added or updated according changes>

**Attached related issue:** (Y/N) <Please add link to related issue here>

**Checklist**:
- [ ] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [ ] Checked that applied changes work as expected

### CODE REVIEW

**Review by merlynomsft — APPROVED** [review]

File: common-npm-packages/azure-arm-rest-v2/azureAppServiceUtility.ts [hunk 1]
```diff
@@ -114,6 +114,15 @@ export class AzureAppServiceUtility {
...
             token = await this._appService._client.getCredentials().getToken();
             method = "Bearer";
+            // Though bearer AuthN is used, lets try to set publish profile password for mask hints to maintain compat with old behavior for MSDEPLOY. 
+            // This needs to be cleaned up once MSDEPLOY suppport is reomve. Safe handle the exception setting up mask hint as we dont want to fail here.
+            try {      
+                tl.setVariable(`AZURE_APP_MSDEPLOY_${this._appService.getSlot()}_PASSWORD`, publishingCredentials.properties["publishingPassword"], true);
+            }
+            catch (error){
+                // safe handle the exception setting up mask hint
+                tl.debug(`Setting mask hint for publish profile password failed with error: ${error}`);
```
Comment (merlynomsft) [inline common-npm-packages/azure-arm-rest-v2/azureAppServiceUtility.ts:124]: Need to bump the version in task.json and task.loc.json

File: common-npm-packages/azure-arm-rest-v2/azureAppServiceUtility.ts [hunk 2]
```diff
@@ -114,6 +114,15 @@ export class AzureAppServiceUtility {
         if(scmPolicyCheck === false) {
             token = await this._appService._client.getCredentials().getToken();
             method = "Bearer";
+            // Though bearer AuthN is used, lets try to set publish profile password for mask hints to maintain compat with old behavior for MSDEPLOY. 
+            // This needs to be cleaned up once MSDEPLOY suppport is reomve. Safe handle the exception setting up mask hint as we dont want to fail here.
```
Comment (merlynomsft) [inline common-npm-packages/azure-arm-rest-v2/azureAppServiceUtility.ts:118]: Changing a common library is a multistep process.  Version in the library needs to be bumped.  Then merged to generate npm package.  Then dependent tasks need to be updated to update version reference in package.json and package-lock.json (Generated by build)

Comment (dhashrath91) [reply common-npm-packages/azure-arm-rest-v2/azureAppServiceUtility.ts:118]: Oh I see.. I was wondering why my changes did not get picked up. understood now.

**Review by manolerazvan — APPROVED** [review]

**Comment by onetocny** [comment]
@microsoft-github-policy-service agree company="Microsoft"

**Pull request closed without merging**